src/sessionization.py

```diff
@@ -20,90 +20,80 @@ def time_diff(dt, dt0):
-    # Read in the weblog file
-    with open(weblog_file) as f1, open(output_file, "a") as f2:
-        f1.readline()
-        # Read in first ip event and initialize first ip session
-        line = f1.readline()
-        ip, dt, date, time = read_line(line)
-        dt0 = dt
-        time_count = 0
-        event_index = 1
-        mod_value = time_count % inact_period
-        ip_dict[ip] = [date, time, dt, date, time, dt, 1, event_index]
+    ip, dt, date, time = pair
+    if ip not in ip_dict:
+        # Initialize ip in this session
+        ip_dict[ip] = [[date, time, dt], [date, time, dt, mod_value], 1, event_index]
         temp.add(ip)
-        for line in f1:
-            # Read in next ip event
-            ip, dt, date, time = read_line(line)
-            event_index += 1
-            diff = time_diff(dt, dt0)
-            if diff > time_count:
-                # For every second, evoke inactive sessions and write to file
-                clean_active_log(mod_value)
-                active_log[mod_value] = temp
-                if ip in ip_dict:
-                    # pop ip from last active session log
-                    last_time_count = time_diff(ip_dict[ip][5], dt0)
-                    last_mod = last_time_count % inact_period
-                    active_log[last_mod].remove(ip)
-                time_count = diff
-                mod_value = time_count % inact_period
-                temp = set()
-            # Update ip dictionary for active sessions
-            # Add ip in updated current active session log with event index
-            if ip not in ip_dict:
-                # Initialize ip in this session
-                ip_dict[ip] = [date, time, dt, date, time, dt, 1, event_index]
-                temp.add(ip)
-            else:
-                # Update current session's last update time and event count
-                ip_dict[ip][3] = date
-                ip_dict[ip][4] = time
-                ip_dict[ip][5] = dt
-                ip_dict[ip][6] += 1
-                temp.add(ip)
-        # Deal with the remaining ip sessions
-        remaindt = set()
-        remaindt.update(active_log[mod_value])
-        active_log[mod_value] = temp
-        for i in range(0, inact_period):
-            remaindt.update(active_log[i])
-        for ip in sorted(remaindt, key=lambda x: ip_dict[x][7]):
-            output_session(ip)
+    else:
+        # Update current session's last update time and event count
+        ip_dict[ip][1] = [date, time, dt, mod_value]
+        ip_dict[ip][2] += 1
+        temp.add(ip)
+    return ip_dict, temp
+
+def process_weblog(f1, f2, ip_dict, temp, active_log, inact_period):
+    """ Process Weblog
+    """
+    f1.readline()
+    # Read in first ip event and initialize first ip session
+    line = f1.readline()
+    ip, dt, date, time = read_line(line)
+    dt0 = dt
+    time_count = 0
+    event_index = 1
+    mod_value = time_count % inact_period
```

Comment:
- perhaps a more descriptive name for ```mod_value```
- what does it do? rather than what it is!